frontends/p4/typeChecking/typeUnification.cpp

```diff
@@ -382,8 +384,7 @@ bool TypeUnification::unify(const IR::Node* errorPosition,
 
         return true;
     } else if (dest->is<IR::Type_Declaration>() && src->is<IR::Type_Declaration>()) {
-        bool canUnify = typeid(dest) == typeid(src) &&
-                dest->to<IR::Type_Declaration>()->name == src->to<IR::Type_Declaration>()->name;
+        bool canUnify = typeid(dest) == typeid(src);
```

Comment:
@mbudiu-vmw  Can you take a look at this line and see if removing the comparison on name makes sense to you?

The check fails for default-package-argument.p4, which it tries to compare M/8 with empty_t and the name of these two types does not match.